homeassistant/components/hue.py

```diff
@@ -0,0 +1,245 @@
+"""
+This component provides basic support for the Philips Hue system.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/hue/
+"""
+import json
+import logging
+import os
+import socket
+
+import voluptuous as vol
+
+from homeassistant.components.discovery import SERVICE_HUE
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import CONF_FILENAME, CONF_HOST
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers import discovery
+
+REQUIREMENTS = ['phue==1.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "hue"
+SERVICE_HUE_SCENE = "hue_activate_scene"
+
+CONF_BRIDGES = "bridges"
+
+CONF_ALLOW_UNREACHABLE = 'allow_unreachable'
+DEFAULT_ALLOW_UNREACHABLE = False
+
+PHUE_CONFIG_FILE = 'phue.conf'
+
+CONF_ALLOW_IN_EMULATED_HUE = "allow_in_emulated_hue"
+DEFAULT_ALLOW_IN_EMULATED_HUE = True
+
+CONF_ALLOW_HUE_GROUPS = "allow_hue_groups"
+DEFAULT_ALLOW_HUE_GROUPS = True
+
+BRIDGE_CONFIG_SCHEMA = vol.Schema([{
+    vol.Optional(CONF_HOST): cv.string,
+    vol.Optional(CONF_FILENAME, default=PHUE_CONFIG_FILE): cv.string,
+    vol.Optional(CONF_ALLOW_UNREACHABLE,
+                 default=DEFAULT_ALLOW_UNREACHABLE): cv.boolean,
+    vol.Optional(CONF_ALLOW_IN_EMULATED_HUE,
+                 default=DEFAULT_ALLOW_IN_EMULATED_HUE): cv.boolean,
+    vol.Optional(CONF_ALLOW_HUE_GROUPS,
+                 default=DEFAULT_ALLOW_HUE_GROUPS): cv.boolean,
+}])
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_BRIDGES, default=[]): BRIDGE_CONFIG_SCHEMA,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+ATTR_GROUP_NAME = "group_name"
+ATTR_SCENE_NAME = "scene_name"
+SCENE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_GROUP_NAME): cv.string,
+    vol.Required(ATTR_SCENE_NAME): cv.string,
+})
+
+CONFIG_INSTRUCTIONS = """
+Press the button on the bridge to register Philips Hue with Home Assistant.
+
+![Location of button on bridge](/static/images/config_philips_hue.jpg)
+"""
+
+
+def setup(hass, config):
+    """Set up the Hue platform."""
+    config = config.get(DOMAIN)
+    if config is None:
+        config = {}
+
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    discovery.listen(
+        hass,
+        SERVICE_HUE,
+        lambda service, discovery_info:
+        bridge_discovered(hass, service, discovery_info))
+
+    bridges = config.get(CONF_BRIDGES, [])
+    if len(bridges) == 0:
+        _LOGGER.info("No bridges found in configuration")
+        return True
```

Comment:
Yeah, I realized that but I thought it might be nice to print something. But I'm not sure it's really useful--after playing with this I'd just recommend letting discovery take care of things.